addons/mail/models/res_partner.py

```diff
@@ -60,7 +60,14 @@ def find_or_create(self, email, assert_valid_email=False):
                 if partners:
                     return partners
 
-        return super(Partner, self).find_or_create(email, assert_valid_email=assert_valid_email)
+        # We don't want to call super to avoid searching twice on the email
+        # Especially when the search email =ilike cannot be as efficient as
+        # a search on email_normalized with a btree index
+        # If you want to override find_or_create your module should depends on mail
+        create_values = {self._rec_name: parsed_name or parsed_email}
+        if parsed_email:  # keep default_email in context
```

Comment:
that inline comment is weird (as in, written backwards) - but I know you did not write it ;-)
```suggestion
        if parsed_email:  # otherwise keep default_email in context
```